topostats/io.py

```diff
@@ -149,16 +159,100 @@ def _load_jpk(self) -> None:
             LOGGER.error(f"[{self.filename}] : {exception}")
         return jpk
 
+    # uncomment below func and import libasd to use asd formats
+    #def load_asd(self) -> tuple:
+    #    """Extract image and pixel to nm scaling from .asd files.
+
+    #    Returns
+    #    -------
+    #    tuple(np.ndarray, float)
+    #        A tuple containing the image and its pixel to nanometre scaling value.
+    #    """
+    #    try:
+    #        scan = libasd.read_asd(self.img_path)
+    #        print(type(scan))
+    #        LOGGER.info(f"[{self.filename}] : Loaded image from : {self.img_path}")
+    #    except FileNotFoundError:
+    #        LOGGER.info(f"[{self.filename}] File not found : {self.img_path}")
+    #    try:
+    #        # libasd docs seem like there is only 2 channels i.e. help(scan)
+    #        # libasd.Header_v0 uses "type" and _v1 uses "kind"
+    #        ch1_name = str(scan.header.data_type_1ch).split(".")[1]
+    #        ch2_name = str(scan.header.data_type_2ch).split(".")[1]
+    #    except AttributeError:
+    #        ch1_name = str(scan.header.data_kind_1ch).split(".")[1]
+    #        ch2_name = str(scan.header.data_kind_2ch).split(".")[1]
+    #    try:
+    #        if self.channel == ch1_name:
+    #            channel_data = scan.channels[0]
+    #        elif self.channel == ch2_name:
+    #            channel_data = scan.channels[1]
+    #        else:
+    #            raise ValueError
+    #        LOGGER.info(f"[{self.filename}] : Extracted channel {self.channel} with {len(channel_data)} frames")
+    #        images = [channel_data[i].image() for i in range(len(channel_data))]
+    #    except ValueError:
+    #        LOGGER.info(
+    #            f"[{self.filename}] : {self.channel} not found in {self.suffix} channel list: [{ch1_name}, {ch2_name}"
+    #        )
+    #        raise ValueError(
+    #            f"[{self.filename}] : {self.channel} not found in {self.suffix} channel list: [{ch1_name}, {ch2_name}"
+    #        )
+
+    #    return (images, self._asd_px_to_nm_scaling(scan))
+
+    # uncomment below func and import libasd to use asd formats
+    #def _asd_px_to_nm_scaling(self, scan: Union[libasd.Data2ch_v0, libasd.Data2ch_v1]) -> float:
+    #    """Extracts the pixel to nanometre scaling value from an .asd file object.
+
+    #    Parameters
+    #    ----------
+        scan: Union[libasd.Data2ch_v0, libasd.Data2ch_v1]
+    #        The object resulting from loading the filepath via the libasd library.
+    #    Returns
+    #    -------
+    #    px_to_m_scaling: float
```

Comment:
Typo here and other places in this function `px_to_m_scaling` > `px_to_nm_scaling`.

However, to be consistent across all other code I think its better to stick with `pixel_to_nm_scaling` as used in `Filters` / `Grains` / `GrainStats` classes). 

They're all the same thing and should therefore have the same name, it makes it easier when using keywords and avoids the possibility that some may think they have different meanings/values.

Reply:
It was in anticipation of going back to the SI units after this task was finished but you're right. Try and keep the things on one branch the same.